lib/std/leb128.zig

```diff
@@ -15,10 +16,8 @@ pub fn readULEB128(comptime T: type, reader: anytype) !T {
     while (group < max_group) : (group += 1) {
         const byte = try reader.readByte();
 
-        const ov = @shlWithOverflow(@as(U, byte & 0x7f), group * 7);
-        if (ov[1] != 0) return error.Overflow;
-
-        value |= ov[0];
+        const mask = try math.shlExact(U, byte & 0x7f, group * 7);
+        value |= mask;
```

Comment:
My understanding is that Andrew prefers to use builtins when available. These should instead be changed to use the destructuring syntax #498:
```suggestion
        const mask, const overflow = @shlWithOverflow(@as(U, byte & 0x7f), group * 7);
        if (overflow != 0) return error.Overflow;

        value |= mask;
```

Reply:
Hmm, I am confused. I feel like that would make sense if you were using `overflow` for bit stuff. However, we seem to be just checking if there wasn't an overflow. Therefore, does it not make much more sense to use the "zig" approach with `try`?

As a side note, this also bypasses the need for prerequisite knowledge for how overflow bits behave (even if they seem simple to some people).